### DISCUSSION TITLE
Bot leveling without players in proximity

### DISCUSSION THREAD

**Discussion by MrRageface** [mod-playerbots/mod-playerbots#458] [Q&A]
Is there a way to have the bots continue to level/quest as normal without having players in proximity? In my testing I noticed when I would teleport to an area with a bot they would be AFK and once it was "aware" of me it would go to killing mobs etc. Is there a way they can continue to grind/quest regardless of players being in the area or is this intentionally not set up for some reason? Or maybe I have overlooked the setting/ did not fully understand the description for such a setting?   Thank you

**Comment by Dreathean** [comment]
So are all of your bots not leveling up at all? Because mine definitely level up without me in proximity, that is how they should be operating. I do not think there is any setting that adjusts this either.

**Reply by MrRageface** [reply]
Correct, as long as I am around them they will function as intended, however when away they AFK. I plan to rebuild the server from scratch again this evening and see if it continues to happen. I thought at first it was a resource issue but I've given the server leagues more than it ever tries to utilize. I wanted to see if there was a known setting that would cause this issue that I may have inadvertently changed in the meantime. 

Edit: And I want to add that it is an immediate change. I can appear in front of a bot, it will go active, teleport to another area then instantly back and the bot will be "Away" sitting just that fast.

Edit 2: I was sitting watching some to see how far they would level with me in the area, 4 of them leveled to 2, then ran far enough away and went AFK.

Edit 3: Rebuilding from scratch has solved the issue, or corrected the problems I caused during the prior at least. I'm curious if someone will know what could have caused it settings-wise or if it was just a fluke in the initial set up.

**Comment by Mr-Deadbeat** [comment]
this might have to do with this setting in worldserver.conf try switching it on if it is not "SetAllCreaturesWithWaypointMovementActive = 1"
the core has those set at default "0" zero so once a player leaves the grid everything stops in it's tracks NOTE!! this will gobble up way more ram especially when you have a lot of bots active so keep that in mind i run server with 32gb ram just to have overhead so far server has at the most used up around 7gb ram after days of server being online i have no doubt it will increase over time once more features gets added to the playerbots my biggest concern is if it will run on older server hardware later on

currently i run the server on an ancient supermicro with dual xeons and it is starting to show it's age ;)

**Comment by hermensbas** [comment]
I rather they dont eat resources when nobody is around tbh.
